### PR TITLE
feat: support i18n for extension. currently only supports English and Chinese.

### PR DESCRIPTION
**Pull request by axetroy**
Now only support configuration and command.

There are some output message still using english.

If you agree with this, I will finish the rest.

refs:

https://github.com/shanalikhan/code-settings-sync/issues/581

### CODE REVIEW

**Comment by shanalikhan** [comment]
This is good.
Thanks !

>If you agree with this, I will finish the rest.

What you will be adding more.
just send me and let me know i will accept and release this in newer version

**Comment by axetroy** [comment]
@shanalikhan

I am trying to translate `updateSetting` command, please check it out and see what can we improve

**Comment by axetroy** [comment]
I am thinking about that should I translate the console message?

console message is only for development, not for user.

so we don't need to translate it.

for example:

https://github.com/shanalikhan/code-settings-sync/blob/3aa9e6c77a6c2ab83f147a73eed926633309e90e/src/commons.ts#L136

**Comment by shanalikhan** [comment]
> I am planning to translate the extension configuration、command、status bar message、information message

This is great.

We dont need to translate the console messages.
Yes i have checked the `updateSetting` title in english and chinese.

So we will be focusing on English and Chinese only for now.

It would be great if you can create new page in Wiki and write some documentation for users to allow their language support for this extension.
What keys they need to localize and what file name should they create.

https://github.com/shanalikhan/code-settings-sync/wiki

File: src/extension.ts [hunk 1]
```diff
@@ -348,7 +349,7 @@ export async function activate(context: vscode.ExtensionContext) {
                             }
                         }
                         else {
-                            console.log(gistName + " key in response is empty.");
+                            console.log(localize('cmd.downloadSettings.warning.responseEmpty', gistName));
```
Comment (shanalikhan) [inline src/extension.ts]: we can exclude Console Log

File: src/extension.ts [hunk 2]
```diff
@@ -417,7 +418,7 @@ export async function activate(context: vscode.ExtensionContext) {
                     }
                 }
                 else {
-                    Commons.LogException(res, "Sync : Unable to Read Gist.", true);
+                    Commons.LogException(res, localize("cmd.downloadSettings.error.readGistFail"), true);
```
Comment (shanalikhan) [inline src/extension.ts]: and this one

**Comment by axetroy** [comment]
@shanalikhan

I have updated a lot for this. it should be almost done.

thanks for your time to review!

here is the document to show how to support their language:

https://github.com/shanalikhan/code-settings-sync/wiki/How-to-support-my-language%3F

**Comment by shanalikhan** [comment]
@axetroy 
It would be great if you can you check the `v3.2` branch. the localization isnt working at the moment. Release is pending due to this.

Thanks

**Comment by axetroy** [comment]
@shanalikhan

Hi.

I have tried `v3.2` in `vscode@1.28.2`

Japanese has worked.

**Comment by shanalikhan** [comment]
Thanks
Its working now

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**